R/compute_age_years.R

```diff
@@ -57,6 +59,13 @@ compute_age_years <- function(age,
     ))
   }
 
+  for (i in seq_along(age_unit)) {
+    if (is.na(age_unit[i])) {
+      age[i] <- NA
+      age_unit[i] <- "years"
+    }
+  }
```

Comment:
A vectorized function like `if_else()` should be used instead of looping over the entries (for performance reasons). For example:
```
time_length(
    duration(age,
      units = if_else(is.na(age_unit), "years", tolower(age_unit))
    ),
    unit = "years"
  )
```

Reply:
But if the units are missing while the age is not, then the above code will not work right? This is why I wrote the loop that also sets the age to NA in the case that the unit is missing.

Reply:
I see. Usually this case should not occur but I agree that it may be safer to return `NA` in this case.
You could use
```
time_length(
    duration(if_else(is.na(age_unit), NA_real_, age),
      units = if_else(is.na(age_unit), "years", tolower(age_unit))
    ),
    unit = "years"
  )
```
but I am not sure if this is better.

Reply:
i do like your fragment better though so we could go with something like that instead? however the code you shared doesn't seem to work for me - it doesn't like the first `if_else` being conditioned on `age_unit` but then affecting the `age` value.

Reply:
Strange, for me it is working. What error message do you get?

Reply:
This fails:
```
  age_input <- c(240, 360, 480, NA)
  age_unit_input <- "MONTHS"

    compute_age_years(
      age_input,
      age_unit_input
    )
```
with error message : Error in (function (cond)  :  error in evaluating the argument 'x' in selecting a method for function 'time_length': `false` must be length 1 (length of `condition`), not 4.

Reply:
I guess because age_unit can be of length 1 when age is a vector.

Reply:
I see. I think we need to expand `age_unit_input` if it has length 1. This case does not work with the for loop as well. Consider for example:
```
 compute_age_years(c(100, 200, NA), age_unit = NA_character_)
[1]  NA 200  NA
```

Reply:
Now implemented.